newIDE/app/src/UI/Slideshow/Slideshow.js

```diff
@@ -98,34 +98,105 @@ const Slideshow = ({
+      // is interacting with the slideshow.
+      if (nextSlideTimeoutId.current) {
+        clearTimeout(nextSlideTimeoutId.current);
+        nextSlideTimeoutId.current = null;
+      }
+
+      setCurrentSlide(currentSlide === 0 ? items.length - 1 : currentSlide - 1);
     },
-    [items]
+    [items, currentSlide]
   );
 
   const handleRightArrowClick = React.useCallback(
     () => {
       if (!items || items.length === 1) return;
 
-      setCurrentSlide(currentSlide =>
-        currentSlide === items.length - 1 ? 0 : currentSlide + 1
-      );
+      // Clear the timeout to avoid changing the slide while the user
+      // is interacting with the slideshow.
+      if (nextSlideTimeoutId.current) {
+        clearTimeout(nextSlideTimeoutId.current);
+        nextSlideTimeoutId.current = null;
+      }
+
+      setCurrentSlide(currentSlide === items.length - 1 ? 0 : currentSlide + 1);
+    },
+    [items, currentSlide]
+  );
+
+  React.useEffect(
+    () => {
+      nextSlideTimeoutId.current = setTimeout(() => {
+        handleRightArrowClick();
+      }, 5000);
+      return () => {
+        clearTimeout(nextSlideTimeoutId.current);
+        nextSlideTimeoutId.current = null;
+      };
     },
-    [items]
+    // The function depends on the currentSlide,
+    // which allows to restart the timeout when the slide changes.
+    [handleRightArrowClick]
   );
 
-  useInterval(() => {
-    handleRightArrowClick();
-  }, 5000);
+  const handleOverImage = React.useCallback(
+    () => {
+      // If the user was going out just before, cancel the timeout.
+      if (outImageTimeoutId.current) {
+        clearTimeout(outImageTimeoutId.current);
+        outImageTimeoutId.current = null;
+      }
+      if (isOverImage) return;
+      setIsOverImage(true);
+    },
+    [isOverImage]
+  );
+
+  const handleOutImage = React.useCallback(
+    () => {
+      // If this event is triggered multiple times, there already is a timeout
+      // so just return.
+      if (!isOverImage || outImageTimeoutId.current) return;
+      outImageTimeoutId.current = setTimeout(() => {
+        setIsOverImage(false);
+      }, 1000);
+      return () => {
+        clearTimeout(outImageTimeoutId.current);
+        outImageTimeoutId.current = null;
+      };
+    },
+    [isOverImage]
+  );
+
+  React.useEffect(
+    () => {
+      const containerElement = containerRef.current;
+      // It's important to wait for the items, so that the listeners are
+      // created when the carousel is actually ready.
+      if (!containerElement || !items) return;
+
+      // Add event listeners on component mount. There is no need to
+      // remove them with a cleanup function because this element
+      // does not change and they will be destroyed when the element is
+      // removed from the DOM.
+      containerElement.addEventListener('mouseover', handleOverImage);
+      containerElement.addEventListener('mouseleave', handleOutImage);
```

Comment:
Why don't you use onMouseEnter and onMouseLeave callbacks from react on the div?

Reply:
btw! prefer pointer events: onPointerOver/onPointerLeave or pointerover pointerleave

Reply:
Yeah forgot I could use the callbacks from react, I've switched that 👍